compiam/io.py

```diff
@@ -2,29 +2,56 @@
 import csv
 import json
 import yaml
+import pickle
 
-import compiam.utils
+import numpy as np
+
+from compiam import utils
 
 def write_csv(data, out_path, header=None):
-    """Writing two dimensional data into a file (.csv)
+    """Writing multi-dimensional data into a file (.csv)
 
     :param data: the data to write
     :param output_path: the path where the data is going to be stored
     
     :returns: None
     """
-    D = list(zip(*data))
+    data = np.array(data)
     with open(out_path, "w") as f:
-        writer = csv.writer(f)
+        writer = csv.writer(f, delimiter=",")
         if header:
-            assert len(header) == len(D[0]), "Header and row length mismatch"
+            if len(header) != len(data[0, :]):
+                raise ValueError("Header and row length mismatch")
             writer.writerow(header)
-        for row in D:
-            writer.writerow(row)
+        writer.writerows(data)
+
+
+def read_csv(file_path):
```

Comment:
Hmm this function is a little weird. Seems like it is expecting all data to be floats and all header to be strings. As such I would not call it read_csv since it is not generalised. I think you can achieve what you have done here (loading a csv to a numpy array) using ` np.genfromtxt(filepath, delimiter=',')`

This does not load the header. But when would you want the header in a numpy array?

Reply:
If we want the header I would use a pd.DataFrame.... which uses np in the background!

Reply:
well, I guess we could use `np.genfromtxt` and then remove nans...